src/main/java/com/redhat/devtools/intellij/tektoncd/actions/DeleteAction.java

```diff
@@ -32,49 +32,63 @@
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath[] path, Object[] selected, Tkn tkncli) {
+        ParentableNode<? extends ParentableNode<NamespaceNode>>[] elements = Arrays.stream(selected).map(item -> getElement(item)).toArray(ParentableNode[]::new);
         int resultDialog = UIHelper.executeInUI(() -> {
-            String kind = element.getClass().getSimpleName().toLowerCase().replace("node", "");
-            return Messages.showYesNoDialog("Are you sure you want to delete " + kind + " " + element.getName() + " ?",
-                    "Delete " + element.getName(),
+            String name = "";
+            String dialogText = "";
+            if (elements.length == 1) {
+                name = elements[0].getName();
+                dialogText = elements[0].getClass().getSimpleName().toLowerCase().replace("node", "") + " " + name + " ?";
+            } else {
+                dialogText = "the following items?\n";
+                for (ParentableNode element: elements) {
+                    dialogText += element.getName() + "\n";
+                }
+            }
+
+            return Messages.showYesNoDialog("Are you sure you want to delete " + dialogText,
+                    "Delete " + name,
                     null
             );
         });
 
         CompletableFuture.runAsync(() -> {
-            try {
-                if (resultDialog == Messages.OK) {
-                    String namespace = element.getParent().getParent().getName();
-                    if (element instanceof PipelineNode) {
-                        tkncli.deletePipeline(namespace, element.getName());
-                    } else if (element instanceof ResourceNode) {
-                        tkncli.deleteResource(namespace, element.getName());
-                    } else if (element instanceof TaskNode) {
-                        tkncli.deleteTask(namespace, element.getName());
-                    } else if (element instanceof  ClusterTaskNode) {
-                        tkncli.deleteClusterTask(element.getName());
-                    } else if (element instanceof ConditionNode) {
-                        tkncli.deleteCondition(namespace, element.getName());
-                    } else if (element instanceof TriggerTemplateNode) {
-                        tkncli.deleteTriggerTemplate(namespace, element.getName());
-                    } else if (element instanceof TriggerBindingNode) {
-                        tkncli.deleteTriggerBinding(namespace, element.getName());
-                    } else if (element instanceof  ClusterTriggerBindingNode) {
-                        tkncli.deleteClusterTriggerBinding(namespace, element.getName());
-                    } else if (element instanceof  EventListenerNode) {
-                        tkncli.deleteEventListener(namespace, element.getName());
+            if (resultDialog == Messages.OK) {
+                Arrays.stream(elements).forEach(element -> {
```

Comment:
Can be rewritten more simply with:

`for(ParentableNode<? extends ParentableNode<NamespaceNode>> element : elements)`